plugin.rb

```diff
@@ -612,4 +612,21 @@ class ::ListableTopicSerializer
 
     options[:refresh_stream] = true if old_category_allows != new_category_allows
   end
+
+  add_to_class(:composer_messages_finder, :check_topic_is_solved) do
+    return unless SiteSetting.solved_enabled &&
+                  !SiteSetting.disable_solved_education_message &&
+                  replying? &&
+                  @topic.present? &&
+                  !@topic.private_message? &&
+                  @topic.custom_fields["accepted_answer_post_id"].present?
```

Comment:
Would be preferable to have this as an `if` instead of an `unless`...I find it pretty hard to read and generally we try to avoid `unless`

```ruby
return if !SiteSetting.solved_enabled || SiteSetting.disable_solved_education_message
return if !replying || @topic.blank? || @topic.private_message?
return if @topic.custom_fields["accepted_answer_post_id"].blank?
```